rust/src/http2/detect.rs

```diff
@@ -861,6 +861,8 @@ pub extern "C" fn rs_http2_tx_add_header(
     let slice_value = build_slice!(value, value_len as usize);
     if slice_name == "HTTP2-Settings".as_bytes() {
         http2_tx_set_settings(state, slice_value)
+    } else if slice_name == "Host".as_bytes() {
```

Comment:
I'm not sure if a header key like "Host" is case sensitive, so would a case-insensitive check make sense here?

Reply:
Looks relevant to me, adding it